docs/source/tutorials/decorator.rst

```diff
@@ -0,0 +1,181 @@
+(such as a function) without modifying its structure: it takes a function as an 
+argument, adds some functionality (could be after and/or before), and returns it.
+
+Its basic syntax is to write :code:`@decorator_name` with its optional
+arguments at the top of a function definition, just as is shown below:
+
+.. code-block:: python
+
+   @decorator(**kwargs)
+   def method(self):
+      return
+      
+In some specific cases (keep reading), this syntax might vary, and you won't be
+able to use the :code:`@` on top of the definition, the method has to be decorated
+manually in the :code:`__init__` function, just as is shown below:
+
+.. code-block:: python
+      
+  def __init__(self):
+    # ...
+    self.method = su.loading_button(alert=w_alert_name, button=w_button_name)(self.method)
+    
+    # ...
+  def method(self):
+     do_something()
+
+.. note:: 
+   
+   Not all the decorators receive arguments, as we will see
+   in the following sections, those are optional and will depend on
+   the decorator objectives.
+
+Decorator types
+===============
+
+In the sepal_ui we have two types of decorators:
+
+- Interactive: those that will trigger traits over objects, and
+- Validation: those that will validate the minimum requirements to 
+run a method/function.
+
+.. note:: This classification is only valid for this tutorial scope and is not
+   applicable for external use.
+
+Interactive
+
+The interactive decorators are intended to be used inside a class 
+inheriting from ipyvuetify widgets (such the SepalWdigets). The
+purpose of them is to trigger an event before, and after the execution
+of a method.
+
+- :code:`loading_button`: is used to capture :code:`Exceptions` and
+  :code:`warnings` from the method and display them into the alert widget, as well as
+  toggle loading the button which trigger the event.
+  
+- :code:`switch`: is used to switch (on/off) any boolean parameter of a ipyvuetify
+  or SepalWidget class.
+
+Validation
+
+The validation decorators are useful when you want to perform some test
+prior the use of a method or a function, the decorator will perform a validation
+tests and if everything is ok, the function/method is executed, otherwise an 
+exception is raised.
+
+- :code:`need_ee`: is used to validate and check if the object requires EE binding, and
+  it will trigger an exception if the connection is not possible.
+
+
+Use cases
+=========
+
+Imagine that you are developing a tile that requires to be connected to your GEE
+account, request your root asset id's and fill up a selection widget. To improve the
+user experience, you also want the following:
+
+- Validate if the SEPAL user is connected to Google Earth Engine.
+- Create an alert and display errors (if there is any).
+- Turn on/off the :code:`loading` and :code:`disabled` parameters of the selection 
+widget while the process is running.
+
+Let's import the required modules. All the decorators are stored in the utils module.
+
+.. code:: python
```

Comment:
prefer the `code-block` command instead of the simpler `code`
It embeds more funcs (as copy-pasting, custom coloring etc...)